lib/content-negotiation.js

```diff
@@ -0,0 +1,93 @@
+// Copyright IBM Corp. 2016. All Rights Reserved.
+// Node module: strong-error-handler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+var accepts = require('accepts');
+var debug = require('debug')('strong-error-handler:http-response');
+var parseParam = require('qs').parse;
+var parseUrl = require('parseurl');
+var sendJson = require('./send-json');
+var sendHTML = require('./send-html');
+
+module.exports = contentNegotiation;
+
+/**
+ * Handles req.accepts and req.query._format and options.defaultTypes
+ * to resolve the correct operation
+ *
+ * @param req request object
+ * @param {Object} options options of strong-error-handler
+ * @returns {Function} Opeartion function with signature `fn(res, data)`
+ */
+function contentNegotiation(req, options) {
+  options = options || {};
+
+  var DEFAULT_SUPPORTED_TYPES = [
+    'application/json', 'application/javascript', 'application/xml',
+    'text/javascript', 'text/html', 'text/xml',
+    'json', 'xml', 'html',
+  ];
+  var defaultType = options.defaultType || 'json';
+
+  // checking if user provided defaultType is supported
+  if (options.defaultType &&
+    DEFAULT_SUPPORTED_TYPES.indexOf(options.defaultType) > -1) {
+    debug('Accepting options.defaultType `%s`', options.defaultType);
+    defaultType = options.defaultType;
+  }
```

Comment:
I think it's more important to print a message (console.warn?) when the default type is not supported, so that the user knows if we are ignoring that setting.